src/trackers/uniqueKeysTracker.ts

```diff
@@ -45,7 +51,7 @@ export function uniqueKeysTrackerFactory(
       
       if (uniqueTrackerSize >= cacheSize) {
         log.warn(`${LOG_PREFIX_UNIQUE_KEYS_TRACKER}The UniqueKeysTracker size reached the maximum limit`);
-        senderAdapter.recordUniqueKeys(uniqueKeysTracker);
+        senderAdapter && senderAdapter.recordUniqueKeys(uniqueKeysTracker);
```

Comment:
Since `senderAdapter` should be required, you can do simply `senderAdapter.recordUniqueKeys(uniqueKeysTracker);`

Reply:
this will be changed to reuse impressionsCountCache in future commits